terraform/pkg/terraformBinary/binary.go

```diff
@@ -0,0 +1,85 @@
+package terraformbinary
+
+import (
+	"context"
+	"os"
+
+	version "github.com/hashicorp/go-version"
+	product "github.com/hashicorp/hc-install/product"
+	releases "github.com/hashicorp/hc-install/releases"
+	tfexec "github.com/hashicorp/terraform-exec/tfexec"
+	log "github.com/rs/zerolog/log"
+)
+
+func installTerraform(binary_dir string, terraform_version string) (string, error) {
+	log.Debug().Msgf("Trying to install terraform binary version %s", terraform_version)
+	installer := &releases.ExactVersion{
+		Product:    product.Terraform,
+		Version:    version.Must(version.NewVersion(terraform_version)),
+		InstallDir: binary_dir,
+	}
+
+	execPath, err := installer.Install(context.Background())
```

Comment:
Warum Background? Das wirkt eher wie TODO 🤔

Reply:
Changed all contexts to TODO. Have to rethink the correct use of context in those function calls.